app/models/node.rb

```diff
@@ -124,7 +124,11 @@ def slug_from_path
   end
 
   def has_a_tag(name)
-    return tags.where(name: name).size.positive?
+    tags.where(name: name).size.positive?
+  end
+
+  def self.hidden_response_node_ids
```

Comment:
I really like this! I'm a little worried about performance. Here we are running some Ruby against every `status = 1` note, right? Take a look at this segment; it's not perfect (it also has to collect /all/ hidden nids before we can exclude them, so years from now if there are thousands, that's a problem), but it does build the condition of having the `hidden:response` tag into the initial query so we aren't running `has_a_tag` on all `status = 1` nodes  in the whole database:
 
https://github.com/publiclab/plots2/blob/444fff0f84af196bb4021c50f68f9ae248ea6449/app/controllers/home_controller.rb#L70-L81

That'll get us to at least as efficient as we were before. Later, it may be worth trying to do more, but since we won't be hitting this query as often (since we're moving it off the dashboard) we don't have to stress too much. I can think of a couple ways to optimize further (either an even more complex SQL query to do it all in one query or... maybe... searching for `hidden:response` nids only within the timeframe of the current set of nodes?) but basically i think we shouldn't worry more about it for now. 😅 

Thanks, Ruth!!!

Reply:
Oh and one more thing -- often if we add a model method, like `node.hidden_response_node_ids`, we add at least a very minimal unit test to protect it, just to confirm that it is providing the most basic response of returning an array of integers, maybe. Could be worthwhile here!

Reply:
Thanks, your suggestion to optimize makes sense, it's just occurred to me that the query with the `has_a_tag` method is an N+1 query.. I'll have a look at the one you've shared and use it. I'll be sure to write some tests for it as well.

Reply:
Can we also update the query from:
```
Node.joins(:node_tag) 
   .joins('LEFT OUTER JOIN term_data ON term_data.tid = community_tags.tid') 
   .select('node.*, term_data.*, community_tags.*') 
   .where(type: 'note', status: 1) 
   .where('term_data.name = (?)', 'hidden:response') 
   .collect(&:nid) 
```
To:

```
 Node.joins(:node_tag) 
   .joins('LEFT OUTER JOIN term_data ON term_data.tid = community_tags.tid') 
   .select('node.nid') 
   .where(type: 'note', status: 1) 
   .where('term_data.name = (?)', 'hidden:response') 
   .collect(&:nid)
```

Since the select statement `.select('node.*, term_data.*, community_tags.*') ` gets all the records and we only collect the nids

Reply:
Ah yes, i think that'll work! I forget, does it automatically add conditional columns or do we need to add "type" and "status"?

Reply:
Yes i think we may need to add any column we use in the query... But good idea!

Reply:
I thought the purpose of the select statement is to determine what is returned? I could be wrong but I tested it with this query that selects the `node.flag`...
<img width="1368" alt="Screenshot 2020-12-15 at 19 21 37" src="https://user-images.githubusercontent.com/7622875/102242148-ceff6900-3f0a-11eb-9340-d60d8b4e58dc.png">


It was able to return two nodes I have locally with the tag name `test` and it only returned the node.flag and did not return the node.nid because I did not select it...

Reply:
Hey @jywarren, I've updated the changes for the query and added a new test in this [PR](https://github.com/publiclab/plots2/pull/8800) which should be the first to be merged.